style.lua

```diff
@@ -119,14 +144,31 @@ function filter_tags_way (keyvalues, numberofkeys)
       return filter, keyvalues, polygon, roads
    end
 
-  -- Treat objects with a key in polygon_keys as polygon
+   -- Treat objects with a key in polygon_keys as polygon
    for i,k in ipairs(polygon_keys) do
       if keyvalues[k] then
+         polygontag = 1
+         -- However, if the key/value combination occurs in linestring_values, do not treat the object as polygon
+         for index,tag in pairs(linestring_values) do
+            if k == tag[1] and keyvalues[k] == tag[2] then
+               polygontag = 0
+               break
+            end
+         end
+         if polygontag == 1 then
+            polygon = 1
```

Comment:
@lonvia If I'm not mistaken, setting 'polygon' directly is not quite the same. Consider an object tagged with a polygon key, like building=yes, and a linestring tag, like historic=citywalls. The outer loop will be run at least twice, once for k=building and once for k=historic. The proposed code sets 'polygontag' and 'polygon' to 1 in the loop with k=building. In the loop with k=historic, 'polygontag' will be set to 0 because historic=citywalls is in linestring_values. However, 'polygon' will remain 1. This is not equivalent to setting 'polygon' directly, because in that case, 'polygon' would also be set to 0 in the loop for k=building.

I admit the code is rather hard to read, perhaps there is a cleaner way to accomplish the same.